### PR TITLE
Migration react-flow-v11-to-v12

### PR DESCRIPTION
**Pull request by jamal-khey**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -36,7 +36,7 @@
         "react-beautiful-dnd": "^13.1.1",
         "react-csv-downloader": "^3.1.0",
         "react-dom": "^18.2.0",
-        "react-flow-renderer": "^10.3.17",
+        "reactflow": "^11.11.4",
```
Comment (SlimaneAmar) [inline package.json:39]: 11 or 12 ?

Comment (jamal-khey) [reply package.json:39]: this PR is not finished yet, i started it from https://github.com/gridsuite/gridstudy-app/pull/2300 to continue on the migration

**Comment by SlimaneAmar** [comment]
Tht content of the toolbar is not displayed !

**Review by EstherDarkish — CHANGES_REQUESTED** [review]
I can consistently make it crash :
In a new study :
1) show the minimap
2) create a new node in a new branch, from the root
3) select the new node (which is created on the right of the existing first node)
4) with a right click on the first node, create a new node in a new branch
=> Crash

File: src/components/graph/network-modification-tree-model.ts [hunk 2]
```diff
@@ -219,10 +226,17 @@ export default class NetworkModificationTreeModel {
 
     setCaseName(newCaseName: string) {
         if (this.treeNodes.length > 0 && this.treeNodes[0].data && newCaseName) {
-            this.treeNodes[0].data = {
-                ...this.treeNodes[0].data,
-                caseName: newCaseName,
-            };
+            const nodeWithOldName = this.treeNodes[0];
+            //check if the node we are modifying is a ROOT node
+            if (nodeWithOldName.type === 'ROOT') {
```
Comment (SlimaneAmar) [inline src/components/graph/network-modification-tree-model.ts:231]: Remove constant 'ROOT'

File: src/redux/reducer.ts [hunk 3]
```diff
@@ -352,16 +352,23 @@ export type StudyUpdated = {
...
     globalBuildStatus?: BUILD_STATUS;
-    localBuildStatus?: BUILD_STATUS;
+    description?: string;
     readOnly?: boolean;
-    caseName?: string;
 };
-export type CurrentTreeNode = Node<TreeNodeData> & { id: UUID };
+export type ReactFlowModificationNodeDataType = NodeCommonData & { localBuildStatus?: BUILD_STATUS };
+
+export type ModificationNode = Node<ReactFlowModificationNodeDataType, 'NETWORK_MODIFICATION'> & {
```
Comment (SlimaneAmar) [inline src/redux/reducer.ts:364]: Remove `Type` at thje end of `ReactFlowModificationNodeDataType`

Comment (jamal-khey) [reply src/redux/reducer.ts]: done

File: src/components/graph/network-modification-tree-model.ts (hunk 2 above)
Comment (jamal-khey) [reply src/components/graph/network-modification-tree-model.ts]: done

File: src/components/graph/util/model-functions.ts [hunk 4]
```diff
@@ -7,45 +7,63 @@
...
-        data: data,
-    };
+
+    if (node.type === NodeType.ROOT) {
+        const data: ReactFlowRootNodeDataType = {
+            parentNodeUuid: parentNodeUuid,
+            label: node.name,
+            description: node.description ?? undefined,
+        };
+        return {
```
[the remark this answers is not in the document]
Comment (jamal-khey) [reply src/components/graph/util/model-functions.ts]: done

File: src/components/graph/network-modification-tree-model.ts [hunk 5]
```diff
@@ -219,10 +226,17 @@ export default class NetworkModificationTreeModel {
 
     setCaseName(newCaseName: string) {
         if (this.treeNodes.length > 0 && this.treeNodes[0].data && newCaseName) {
-            this.treeNodes[0].data = {
-                ...this.treeNodes[0].data,
-                caseName: newCaseName,
-            };
+            const nodeWithOldName = this.treeNodes[0];
+            //check if the node we are modifying is a ROOT node
+            if (nodeWithOldName.type === NodeType.ROOT) {
```
Comment (SlimaneAmar) [inline src/components/graph/network-modification-tree-model.ts:231]: use `isRootNode` ?

File: src/redux/reducer.ts [hunk 6]
```diff
@@ -352,16 +352,23 @@ export type StudyUpdated = {
...
 };
-export type CurrentTreeNode = Node<TreeNodeData> & { id: UUID };
+export type ReactFlowModificationNodeData = NodeCommonData & { localBuildStatus?: BUILD_STATUS };
+
+export type ModificationNode = Node<ReactFlowModificationNodeData, 'NETWORK_MODIFICATION'> & {
+    id: UUID;
+};
+
+export type ReactFlowRootNodeData = NodeCommonData & { caseName?: string };
+export type RootNode = Node<ReactFlowRootNodeData, 'ROOT'> & { id: UUID };
```
Comment (SlimaneAmar) [inline src/redux/reducer.ts:369]: 'ROOT' constant ?

File: src/components/graph/network-modification-tree-model.ts [hunk 7]
```diff
@@ -174,29 +179,31 @@ export default class NetworkModificationTreeModel {
...
+                const modificationNodeData = getModificationNodeDataOrUndefined(node);
+                const globalBuildStatus = modificationNodeData?.nodeBuildStatus?.globalBuildStatus;
+                const localBuildStatus = modificationNodeData?.nodeBuildStatus?.localBuildStatus;
+                this.treeNodes[indexModifiedNode] = {
+                    ...nodeToUpdate,
+                    data: {
+                        ...this.treeNodes[indexModifiedNode].data,
+                        label: node.name,
+                        globalBuildStatus: globalBuildStatus,
+                        // @ts-ignore TODO separate update root and update modification ?
```
Comment (SlimaneAmar) [inline src/components/graph/network-modification-tree-model.ts:196]: separate root and modification `node`

**Review by SlimaneAmar — APPROVED** [review]

**Review by EstherDarkish — APPROVED** [review]

File: src/components/graph/network-modification-tree-model.ts (hunk 5 above)
Comment (jamal-khey) [reply src/components/graph/network-modification-tree-model.ts]: done

File: src/redux/reducer.ts (hunk 6 above)
Comment (jamal-khey) [reply src/redux/reducer.ts]: done

File: src/components/graph/network-modification-tree-model.ts (hunk 7 above)
Comment (jamal-khey) [reply src/components/graph/network-modification-tree-model.ts]: ts-ignore removed i separated the updates

**Pull request merged**